### PR TITLE
Add Mocks

### PR DESCRIPTION
**Pull request by nakulgan**
Mocks use [ApiBlueprint](http://apiblueprint.org)
Uses Drakov to run Mocks

### CODE REVIEW

File: util/mocks/package.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "name": "Api Mocks",
```
Comment (dominicumbeer) [inline util/mocks/package.json]: Package name does not include a whitespace. "Api Mocks" => "ApiMocks"

File: util/mocks/Makefile [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env make
+
+.PHONY: serve install
+
+install:
+	npm install
+
+run:
+	drakov -f ./api.md
```
Comment (dominicumbeer) [inline util/mocks/Makefile]: `drakov` is normally not installed globally. This line would solve the problem:

`./node_modules/.bin/drakov -f ./api.md`

File: util/mocks/api.md [hunk 3]
```diff
@@ -0,0 +1,733 @@
...
+          "id": 3,
+          "firstname": "Jack",
+          "lastname": "Joker",
+          "email": "need.a@supervillaain.com",
+          "active": 1,
+          "updated_at": "2015-03-11T08:40:51.620Z",
+          "created_at": "2015-02-24T08:40:51.620Z"
+        }
+
+## User [/users/{id}]
```
Comment (dominicumbeer) [inline util/mocks/api.md]: Just a comment and question: In the defined case we could call `<endpoint>/users/5` and we'd get an user as result which has  an `id` of 3. This is inconsistent! Do you know if there is any way to be more dynamic even in the mock definition?

File: util/mocks/api.md [hunk 4]
```diff
@@ -0,0 +1,733 @@
...
+        }
+
++ Response 201 (application/json)
+
+        {
+          "id": 3,
+          "firstname": "Jack",
+          "lastname": "Joker",
+          "email": "need.a@supervillaain.com",
+          "active": 1,
```
Comment (dominicumbeer) [inline util/mocks/api.md]: According to schema an user shouldn't be activated by default ;)

File: util/mocks/api.md [hunk 5]
```diff
@@ -0,0 +1,733 @@
...
++ Response 201 (application/json)
+
+        {
+          "id": 3,
+          "name": "SuperNormalGuy",
+          "url": "http://www.supernormalguy.com",
+          "language": "de-DE",
+          "active": 0,
+          "user_id": 4,
+          "updated_at": "2014-11-11T08:40:51.620Z",
```
Comment (dominicumbeer) [inline util/mocks/api.md]: `updated_at` and `created_at` should be the same after creation.

**Comment by dominicumbeer** [comment]
What about update functionality? Shall I add it later?

File: util/mocks/api.md [hunk 6]
```diff
@@ -0,0 +1,733 @@
...
++ id - unique campaigns_category id
++ campaign_id
++ category_id
++ updated_at 
++ created_at
+
++ Parameters
+    + id (required, number, `1`)
+
+## campaigns_category Collection [/campaigns_categories]
```
Comment (dominicumbeer) [inline util/mocks/api.md]: Is it good API design to expose relation tables? I would offer these information directly on the corresponding entities (campaign or category). 

Campaign contains already all categories:
```
 {
           "id": 3,
           "type": "cpc",
           "title": "SuperNormalCampaign",
           "start_date": "2015-11-11T08:40:51.620Z",
           "end_date": "2015-08-24T08:40:51.620Z"
           "active": 1,
           "categories": [
                  {....},
                  {....}
           ]
           "updated_at": "2014-11-11T08:40:51.620Z",
           "created_at": "2014-08-24T08:40:51.620Z"
}
```

Additionally we could support filters on the campaign entity itself:
`<endpoint>/campaigns/?categories=1,2,5`

I'm not an expert in REST design but this was my initial idea. What do you think?

**Comment by nakulgan** [comment]
:cactus:

**Pull request closed without merging**